usdt/src/lib.rs

```diff
@@ -12,20 +15,50 @@
 //! };
 //! ```
 //!
+//! Providers and probes may be named in any way, as long as they form valid Rust identifiers. The
+//! names are intended to help understand the behavior of a program, so they should be semantically
+//! meaningful. Probes accept zero or more arguments, data that is associated with the probe event
+//! itself (timestamps, file descriptors, filesystem paths, etc.). The arguments may be specified
+//! as any of the exact bit-width integer types (e.g., `int16_t`) or strings (`char *`s). See
+//! [Data types](#data-types) for a full list of supported types.
+//!
 //! Assuming the above is in a file called `"test.d"`, this may be compiled into Rust code with:
 //!
 //! ```ignore
+//! #![feature(asm)]
```

Comment:
where does this need to appear? mod.rs? lib.rs? main.rs?

Reply:
Good question, I'm not sure yet. I'm running some experiments to find out.

Reply:
After some experimentation, here's what I've found. Looks like the `!#[feature(asm)]` directive must be at the crate root. The story around the proc-macro invocation site is more complex, since it generates macros. There's at least one improvement that we can do (getting the full path of the probe and is-enabled probe in the proc-macro), but practically speaking, `usdt::dtrace_provider!` should also be invoked at the crate root.